### PR TITLE
Adding YOLOFastestV2 full model except loss 

### PR DESCRIPTION
**Pull request by hglee98**
## Description

This PR aims to support YOLOFastestV2 model (#376 ).
However, There are some works left to do. 
- Implement correspoding loss (for now I choose to use existing retinanetloss).
- Load pretrained weights
- Add docs

We recommend to link at least one existing issue for PR. Before your create a PR, please check if there is an issue for this change.  

## Change(s)

- Add YOLOFastestV2 config 
- Implement SuffleNetV2 backbone and block
- Implement LightFPN for YOLOFastestV2 
- Add DWConvBlock for LightFPN 

## Code Formatting

If you PR to either `master` or `dev` branch, you should follow the code linting process. Please check your code with `lint_check.sh` in `./scripts` directory.
For more information, please read the contribution guide in `CONTRIBUTING.md`. 

## Changelog

If your PR is granted to `dev` branch, codeowner will add a brief summary of the change to the **Upcoming Release** section of the [`CHANGELOG.md`](https://github.com/Nota-NetsPresso/netspresso-trainer/blob/master/CHANGELOG.md) file including a link to the PR (formatted in markdown) and a link to your github profile.

For example,

```
- Added a new feature by `@myusername` in [PR 2023](https://github.com/Nota-NetsPresso/netspresso-trainer/pull/2023)
```

Please enable **Allow edits and access to secrets by maintainers** so that our maintainers can update the `CHANGELOG.md`.

### CODE REVIEW

File: config/model/yolo/yolo-fastest-v2-detection.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      params:
+        anchors:
+          - [12,18, 37,49, 52,132]  # P2
+          - [115,73, 119,199, 242,238]  # P3
+        # postprocessor - decode
+        topk_candidates: 1000
+        score_thresh: 0.05
+        # postprocessor - nms
+        nms_thresh: 0.45
+        class_agnostic: False
```
Comment (illian01) [inline config/model/yolo/yolo-fastest-v2-detection.yaml:34]: This should be changed after #470 merged.

File: src/netspresso_trainer/models/op/custom.py [hunk 2]
```diff
@@ -646,6 +646,85 @@ def forward(self, x):
...
+    def channel_shuffle(self, x):
+        b, c, h, w = x.data.size()
+        assert (c % 4 == 0)
+        x = x.reshape(b * c // 2, 2, h * w)
+        x = x.permute(1, 0, 2)
+        x = x.reshape(2, -1, c // 2, h, w)
+        return x[0], x[1]
+
+
+class DWConvBlock(nn.Module):
```
Comment (illian01) [inline src/netspresso_trainer/models/op/custom.py:707]: I think this name makes confusing with existing `SeparableConvLayer`.
How about changing this class name?

**Review by illian01 — CHANGES_REQUESTED** [review]
It looks good, but some minor extra modification is needed after merge #470
Please see details.

File: src/netspresso_trainer/models/op/custom.py (hunk 2 above)
Comment (hglee98) [reply src/netspresso_trainer/models/op/custom.py:707]: Thank you for your suggestion! Let me double-check and see if I can change `DWConvBlock` to an appropriate name or integrate `DWConvBlock` with `SeparableConvLayer` 😊

File: config/model/yolo/yolo-fastest-v2-detection.yaml (hunk 1 above)
Comment (hglee98) [reply config/model/yolo/yolo-fastest-v2-detection.yaml:34]: It makes sense! Thank you for your suggestion!

**Comment by hglee98** [comment]
As we discussed, here are changes that I make. 

- Remove `DWConvBlock` and use `SeparableConvLayer` as a building block to represent it. 
- Add postprocessor parameters section to yolofastest v2 configuration file as merged #470.  

Thank you for your consideration :)

**Review by illian01 — APPROVED** [review]
LGTM!
I will merge this PR after updating minor things.

File: config/model/yolo/yolo-fastest-v2-detection.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    optimizer_path: ~
+  freeze_backbone: False
+  architecture:
+    full: ~ # auto
+    backbone:
+      name: shufflenetv2
+      params: 
+        model_size: 0.5x
+      stage_params: 
+        out_channels: [-1, 24, 48, 96, 192]
```
[the remark this answers is not in the document]
Comment (hglee98) [reply config/model/yolo/yolo-fastest-v2-detection.yaml:18]: Let me check this issue quickly!

[the remark this answers is not in the document]
Comment (hglee98) [reply config/model/yolo/yolo-fastest-v2-detection.yaml:18]: Ah~ yes you're right the stage params are controlled by params.model_size 😅
Thank you for your suggestion!

File: src/netspresso_trainer/models/backbones/experimental/shufflenetv2.py [hunk 4]
```diff
@@ -40,6 +40,12 @@ def __init__(
             self.stage_out_channels = [-1, 24, 176, 352, 704]
         elif model_size == '2.0x': 
             self.stage_out_channels = [-1 ,24, 244, 488, 976]
+        else:
+            raise NotImplementedError(
```
Comment (hglee98) [inline src/netspresso_trainer/models/backbones/experimental/shufflenetv2.py:44]: @illian01 This change implements a check for supported model_size values ('0.5x', '1.0x', '1.5x', '2.0x'). It raises a NotImplementedError if an unsupported value is provided, preventing potential issues downstream and improving user experience by providing clear feedback on valid options.

Comment (illian01) [reply src/netspresso_trainer/models/backbones/experimental/shufflenetv2.py:44]: Thank you for your detailed update.
I agree with it.

**Comment by hglee98** [comment]
@illian01

I've completed a refactoring task for the `ShuffleNetV2` and `ShuffleV2Block` classes to improve their clarity and maintainability.
Could you please review these commits and provide any feedback or questions you may have?

Additionally:
- I've created a separate branch for the implementation of `YOLOFastestLoss`.
- I believe it would be more organized to open a corresponding PR for `YOLOFastestLoss` after this current PR is merged.
- If you have any suggestions or ideas regarding the `YOLOFastestLoss` implementation, please feel free to share them with me!

Looking forward to your review and thoughts on both the refactoring and the upcoming `YOLOFastestLoss` PR.

**Review by illian01 — APPROVED** [review]
Sorry for late review.
I checked all your code, and everything look good to me.
Thank you for your effort 👍
